File: SCT2/trunk/test-plugins/org.yakindu.sct.model.sexec.interpreter.test/test-gen/org/yakindu/sct/model/sexec/interpreter/test/NamedInterfaceAccessTest.java
/**
 * Copyright (c) 2013 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.model.sexec.interpreter.test;
import static junit.framework.Assert.assertTrue;

import org.eclipse.xtext.junit4.InjectWith;
import org.eclipse.xtext.junit4.XtextRunner;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.yakindu.sct.model.sexec.ExecutionFlow;
import org.yakindu.sct.model.sexec.interpreter.test.util.AbstractExecutionFlowTest;
import org.yakindu.sct.model.sexec.interpreter.test.util.SExecInjectionProvider;
import org.yakindu.sct.test.models.SCTUnitTestModels;

import com.google.inject.Inject;
/**
 *  Unit TestCase for NamedInterfaceAccess
 */
@SuppressWarnings("all")
@RunWith(XtextRunner.class)
@InjectWith(SExecInjectionProvider.class)
public class NamedInterfaceAccessTest extends AbstractExecutionFlowTest {
	@Inject
	private SCTUnitTestModels models;
	@Before
	public void setup() throws Exception {
		ExecutionFlow flow = models
				.loadExecutionFlowFromResource("NamedInterfaceAccess.sct");
		initInterpreter(flow);
	}
	@Test
	public void SafeOpenSuccess() throws Exception {
		interpreter.enter();
		interpreter.runCycle();
		raiseEvent("User.numberPressed", 3);
		interpreter.runCycle();
		raiseEvent("User.numberPressed", 7);
		interpreter.runCycle();
		raiseEvent("User.numberPressed", 5);
		interpreter.runCycle();
		assertTrue(isRaised("Safe.open"));
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/NamedInterfaceAccessTest/NamedInterfaceAccessTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "NamedInterfaceAccess.h"

NamedInterfaceAccess handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		namedInterfaceAccess_init(&handle);
	}
};


TEST_F(StatemachineTest, SafeOpenSuccess) {					
	namedInterfaceAccess_enter(&handle);
	namedInterfaceAccess_runCycle(&handle);
	namedInterfaceAccessIfaceUser_raise_numberPressed(&handle, 3l);
	namedInterfaceAccess_runCycle(&handle);
	namedInterfaceAccessIfaceUser_raise_numberPressed(&handle, 7l);
	namedInterfaceAccess_runCycle(&handle);
	namedInterfaceAccessIfaceUser_raise_numberPressed(&handle, 5l);
	namedInterfaceAccess_runCycle(&handle);
	EXPECT_TRUE(namedInterfaceAccessIfaceSafe_israised_open(&handle));
}


